main.go

```diff
@@ -103,8 +114,13 @@ func main() {
 				kustomize.WithResourcePath(*resourceDirInput),
 				kustomize.AdditionalOverlayGenerator(kustomize.NewPatchOverlayGenerator(sd.Spec.Behavior.Engine.Kustomize.Overlays)),
 			)))
+	case Helm3Engine:
+		options = append(options,
```

Comment:
Ok, so it seems like using the kustomize engine is the default, and that's what's returned by the constructor, and then all of the options methods are called, and this one sets the engine as the helm3 engine instead. Is that a common pattern in `go` for setting the value of a field? I suppose I'm not entirely used to it yet.

Reply:
The default could've been a `NopEngine`, too, which simply does nothing. I think that's what we have in other reconcilers, too, instead of opting to use an actual engine. I'll write a simple `NopEngine` and assign that by default.

Also it'd be good to have `default:` in switch statement that just returns error.